autogen/main.tf

```
@@ -36,290 +36,118 @@ resource "random_shuffle" "available_zones" {
 }
 
 locals {
-  kubernetes_version_regional = var.kubernetes_version != "latest" ? var.kubernetes_version : data.google_container_engine_versions.region.latest_master_version
-  kubernetes_version_zonal    = var.kubernetes_version != "latest" ? var.kubernetes_version : data.google_container_engine_versions.zone.latest_master_version
-  node_version_regional       = var.node_version != "" && var.regional ? var.node_version : local.kubernetes_version_regional
-  node_version_zonal          = var.node_version != "" && !var.regional ? var.node_version : local.kubernetes_version_zonal
+  location                    = var.regional ? var.region : var.zones[0]
+  node_locations              = var.regional ? sort(random_shuffle.available_zones.result) : slice(compact(var.zones), 1, length(compact(var.zones)))
+  latest_master_version       = data.google_container_engine_versions.master.latest_master_version
+  latest_node_version         = data.google_container_engine_versions.node.latest_master_version
+  kubernetes_master_version   = var.kubernetes_version != "latest" ? var.kubernetes_version : local.latest_master_version
+  kubernetes_node_version     = var.node_version != "" ? var.kubernetes_version : local.latest_node_version
```

Shouldn't this be var.node_version if it's provided?

```suggestion
  kubernetes_node_version     = var.node_version != "" ? var.node_version : local.latest_node_version
```

---

fixed